problem-gson/src/test/java/org/zalando/problem/GsonProblemTest.java

```diff
@@ -0,0 +1,296 @@
+            assertThat(problem, hasFeature("debit", InsufficientFundsProblem::getDebit, equalTo(-20)));
+        }
+    }
+
+    @Test
+    void shouldDeserializeUnknownStatus() throws IOException {
+        try (final JsonReader reader = getReader("unknown.json")) {
+            final Problem problem = gson.fromJson(reader, Problem.class);
+
+            final StatusType status = problem.getStatus();
+
+            assertThat(status, hasFeature("status code", StatusType::getStatusCode, equalTo(666)));
+            assertThat(status, hasFeature("reason phrase", StatusType::getReasonPhrase, equalTo("Unknown")));
+        }
+    }
+
+    @Test
+    void shouldDeserializeUntyped() throws IOException {
+        try (final JsonReader reader = getReader("untyped.json")) {
+            final Problem problem = gson.fromJson(reader, Problem.class);
+
+            assertThat(problem.getType(), hasToString("about:blank"));
+            assertThat(problem.getTitle(), is("Something bad"));
+            assertThat(problem.getStatus(), hasFeature(StatusType::getStatusCode, is(400)));
+            assertThat(problem.getDetail(), is(nullValue()));
+            assertThat(problem.getInstance(), is(nullValue()));
+        }
+    }
+
+    @Test
+    void shouldDeserializeEmpty() throws IOException {
+        try (final JsonReader reader = getReader("empty.json")) {
+            final Problem problem = gson.fromJson(reader, Problem.class);
+
+            assertThat(problem.getType(), hasToString("about:blank"));
+            assertThat(problem.getTitle(), is(nullValue()));
+            assertThat(problem.getStatus(), is(nullValue()));
+            assertThat(problem.getDetail(), is(nullValue()));
+            assertThat(problem.getInstance(), is(nullValue()));
+        }
+    }
+
+    @Test
+    void shouldDeserializeCause() throws IOException {
+        try (final JsonReader reader = getReader("cause.json")) {
+            final ThrowableProblem problem = gson.fromJson(reader, ThrowableProblem.class);
+
+            assertThat(problem, hasFeature("cause", Throwable::getCause, is(notNullValue())));
+            final DefaultProblem cause = (DefaultProblem) problem.getCause();
+
+            assertThat(cause, is(notNullValue()));
+            assertThat(cause, instanceOf(DefaultProblem.class));
+
+            assertThat(cause, hasFeature("type", Problem::getType, hasToString("https://example.org/expired-credit-card")));
+            assertThat(cause, hasFeature("title", Problem::getTitle, equalTo("Expired Credit Card")));
+            assertThat(cause, hasFeature("status", Problem::getStatus, equalTo(BAD_REQUEST)));
+            assertThat(cause, hasFeature("detail", Problem::getDetail, is("Credit card is expired as of 2015-09-16T00:00:00Z")));
+            assertThat(cause, hasFeature("parameters", Problem::getParameters, hasEntry("since", "2015-09-16T00:00:00Z")));
+        }
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/whiskeysierra/problem/pullRequest?prid=4476243)